### PR TITLE
minor polish to friendly error messages

### PR DESCRIPTION
**Pull request by anurse**
I was helping @kichalla out with something and realized that the friendly error messages needed a small tweak to improve the user story a little. First, we need a way to dump the full stack trace when diagnosing issues, originally I thought a separate environment variable would be good, but I think DNX_TRACE can do the job. Second, we need to handle AggregateException separately and write out the individual exceptions since even single exceptions thrown in async code get wrapped in AEs.

For: RC1

* dump full exception stack trace when DNX_TRACE=1
* dump each exception inside an AggregateException

/cc @JunTaoLuo @troydai @halter73

### CODE REVIEW

File: src/Microsoft.Dnx.Tooling/Program.cs [hunk 1]
```diff
@@ -98,12 +98,26 @@ public int Main(string[] args)
                 throw;
             }
 #else
+            catch(AggregateException aex)
```
Comment (davidfowl) [inline src/Microsoft.Dnx.Tooling/Program.cs]: formatting!

Comment (troydai) [inline src/Microsoft.Dnx.Tooling/Program.cs]: formatting!

File: src/Microsoft.Dnx.Tooling/Program.cs [hunk 2]
```diff
@@ -98,12 +98,26 @@ public int Main(string[] args)
...
+            {
+                foreach(var exception in aex.InnerExceptions)
```
Comment (troydai) [inline src/Microsoft.Dnx.Tooling/Program.cs]: formatting!

File: src/Microsoft.Dnx.Tooling/Program.cs [hunk 3]
```diff
@@ -98,12 +98,26 @@ public int Main(string[] args)
...
-                AnsiConsole.GetError(useConsoleColor: _runtimeEnv.OperatingSystem == "Windows").WriteLine($"Error: {ex.Message}".Red().Bold());
+                DumpException(ex);
                 return 1;
             }
 #endif
         }
+
+        private void DumpException(Exception ex)
+        {
+            AnsiConsole.GetError(useConsoleColor: _runtimeEnv.OperatingSystem == "Windows").WriteLine($"Error: {ex.Message}".Red().Bold());
```
Comment (troydai) [inline src/Microsoft.Dnx.Tooling/Program.cs]: Any change to wrap this line?

**Comment by troydai** [comment]
:shipit: after Travis passes and format fixes.

File: src/Microsoft.Dnx.Tooling/Program.cs [hunk 4]
```diff
@@ -98,12 +98,28 @@ public int Main(string[] args)
...
                 return 1;
             }
 #endif
         }
+
+        private void DumpException(Exception ex)
+        {
+            AnsiConsole
+                .GetError(useConsoleColor: _runtimeEnv.OperatingSystem == "Windows")
+                .WriteLine($"Error: {ex.Message}".Red().Bold());
```
Comment (halter73) [inline src/Microsoft.Dnx.Tooling/Program.cs]: Why ex.Message instead of ex.ToString() like you use for TraceError?

Comment (anurse) [inline src/Microsoft.Dnx.Tooling/Program.cs]: Because we only want to show the short summary message here. `ToString` includes the stack trace

Comment (anurse) [inline src/Microsoft.Dnx.Tooling/Program.cs]: For example, when `DNX_TRACE` is off, we want to see:

```
Error: Something bad happened
```

But with `DNX_TRACE`:

```
Error: Something bad happened
[Error] Full Exception: System.SomethingBadHappenedException: Something bad happend
 at blah.blah.call.stack
 at blah.blah.call.stack
```

Comment (davidfowl) [inline src/Microsoft.Dnx.Tooling/Program.cs]: example?

**Comment by halter73** [comment]
:ship:

**Comment by davidfowl** [comment]
:shipit:

**Pull request closed without merging**